code/modules/antagonists/changeling/powers/fakedeath.dm

```diff
@@ -37,7 +37,7 @@
 	user.regenerate_organs()
 
 /datum/action/changeling/fakedeath/proc/ready_to_regenerate(mob/user)
-	if(user?.mind)
+	if(user?.mind && ishuman(usr))
```

Comment:
```suggestion
	if(user?.mind && ishuman(user))
```

Reply:
Would be better to change the typepath of the arg and then just do `istype(user)`

Reply:
I did try that, but the proc would still go through when headspidered which is what I'm trying to avoid I'm pretty sure that what's saved is the mob/living/user of when you were still a human.

Reply:
Hmm yeah, since this is how the proc gets called 
`addtimer(CALLBACK(src, .proc/ready_to_regenerate, user), LING_FAKEDEATH_TIME, TIMER_UNIQUE)`
 it remembers the arg as user who invoked it first, but could change from human to headspider during the timer. Maybe a better solution would be to pass `usr` instead of `user` into the callback, rather than checking the `usr` in called function? Idk, someone smarter has to confirm

Reply:
Could pass in the mind rather than the mob, so you can do mind.current.

Reply:
is what I just pushed suitable?

Reply:
The problem is because when you pass in the mob, user is a human.
The user then becomes a changeling and the mob gets deleted, nulling the reference to that human.
User will reference a qdeleted thing and could potentially cause hard-dels (bad). This would probably work, but it shouldn't and might not always work if the reference remover is fast enough.
I would recommend updating the parameters of the proc to be `datum/mind/mind` rather than `mob/user` and then changing the addtimer to pass in a mind instead of a mob

Reply:
Alright, gotcha and thanks for the explanation

Reply:
Should be good to go now then